# Core-Collapse Supernovae

In `skysurvey`, Core-Collapse (CC) SNe, including Type II, IIn, IIb, Ib, Ic, and Ic-BL, are pre-built classes that inherit from a common base.

***

## Class structure


Their class structure are defined by:
- `VincenziModels`, which provides the link to Vincenzi et al. (2019) (2019MNRAS.489.5802V) time-series models implemented in `sncosmo`.
- `MultiTemplateTSTransient`, which allows combining multiple templates to model the diversity of observed CC SNe. 
- Each subtype inherits its own parameters: `_KIND`, `_RATE`, `_MAGABS`, and a set of v19-*-corr templates.

## Templates

The different CC SNe classes are associated with time-series templates from Vincenzi et al. (2019), implemented in `sncosmo` under the v19-*-corr name, and stored in `skysurvey` as a `TemplateCollection` object.

Let's take the example of the `SNeII` class for the following sections. The SNeII Transient is already defined and ready to use: 

In [9]:
import skysurvey

snii = skysurvey.SNeII()

The `template` (skysurvey object) is stored here:

In [ ]:
snii.template

The available templates are:

In [3]:
snii.template.names

['v19-asassn14jb-corr',
 'v19-asassn15oz-corr',
 'v19-1987a-corr',
 'v19-1999em-corr',
 'v19-2004et-corr',
 'v19-2007od-corr',
 'v19-2008bj-corr',
 'v19-2008in-corr',
 'v19-2009n-corr',
 'v19-2009bw-corr',
 'v19-2009dd-corr',
 'v19-2009ib-corr',
 'v19-2009kr-corr',
 'v19-2012a-corr',
 'v19-2012aw-corr',
 'v19-2013ab-corr',
 'v19-2013am-corr',
 'v19-2013by-corr',
 'v19-2013ej-corr',
 'v19-2013fs-corr',
 'v19-2014g-corr',
 'v19-2016x-corr',
 'v19-2016bkv-corr']

## Rates

Each pre-computed skysurvey transient has a default volumetric rate, that provide the number of transient expected per year and per Gpc³.

The rate of the `SNeII` is calculated as `CC_RATE` * 0.649, derived from the total CC SNe rate (`CC_RATE` = 1.0e5) based on Perley et al. (2020) (2020ApJ...904...35P) and the relative rate from Vicenzi et al. (2019) (that combines both IIL and IIP types). 

In [4]:
snii.rate

64900.0

The other CC SNe rates are also derived from the relative rate from Vicenzi et al. (2019) multipled by the total CC SNe rate. The `SNeICL` relative rate is defined joining two types, the Ic-BL and Ic-pec types.

| Class | rate (target/yr/Gpc^3) | 
|-------|-----------------------------|
| `SNeII` | 64900 | 
| `SNeIIn` | 4700| 
| `SNeIIb` | 10900| 
| `SNeIb` | 10800|
| `SNeIc` | 7500| 
| `SNeIcBL` | 9700|

## Model

The model is based on the [modeldag](https://github.com/MickaelRigault/modeldag) package.

For the CC SNe classes, the model contains 7 entries. So the generated data will contains at least 7 columns. To display the model, you can directly print the object:

In [12]:
snii

{'redshift': {'kwargs': {'zmax': 0.05, 'rate': 64900.0}, 'as': 'z'},
 't0': {'func': <bound method Generator.uniform of Generator(PCG64) at 0x133B6C040>,
        'kwargs': {'low': 56000, 'high': 56200}},
 'magabs': {'func': <bound method Generator.normal of Generator(PCG64) at 0x133B6C120>,
            'kwargs': {'loc': -17.48, 'scale': 0.7}},
 'magobs': {'func': 'magabs_to_magobs',
            'kwargs': {'z': '@z', 'magabs': '@magabs'}},
 'amplitude': {'func': 'magobs_to_amplitude', 'kwargs': {'magobs': '@magobs'}},
 'radec': {'func': <function random_radec at 0x133b0fba0>,
           'as': ['ra', 'dec'],
           'kwargs': {}},
 'template': {'func': 'draw_template', 'kwargs': {'redshift': '@z'}}}

## Absolute magnitude distribution

The absolute magnitude distribution is assumed to be Gaussian, specified by (location, scatter). The chosen values are derived from the Median R-band magnitude (MR) from the BTS (Bright Transient Survey) for z<0.05.

In [5]:
snii._MAGABS

(-17.48, 0.7)

| Class | `magabs` | 
|-------|-----------------------------|
| `SNeII` | (-17.48, 0.7) | 
| `SNeIIn` | (-18.0, 0.8) | 
| `SNeIIb` | (-17.45, 0.6) | 
| `SNeIb` | (-17.35, 0.53) |
| `SNeIc` | (-17.50, 0.7) | 
| `SNeIcBL` | (-18.12, 0.9) |